server/auth/store_test.go

```diff
@@ -113,12 +113,14 @@ func setupAuthStore(t *testing.T) (store *authStore, teardownfunc func(t *testin
 
 	// The UserAdd function cannot generate old etcd version user data (user's option is nil)
 	// add special users through the underlying interface
-	addUserWithNoOption(as)
+	asImpl, ok := as.(*authStore)
+	require.Truef(t, ok, "addUserWithNoOption: needs an AuthStore implementation")
```

Comment:
Can we add a validation in production code similar to below?
https://github.com/etcd-io/etcd/blob/4abcda49e0b8945fdb17d3db219337caa3ee9a2f/server/storage/schema/auth.go#L41


```suggestion
```